core/state_processor.go

```diff
@@ -409,8 +412,12 @@ func ApplyInnerTransaction(bc *BlockChain, author *common.Address, gp *GasPool,
 		return errors.New("attempt to process non-private transaction from within ApplyInnerTransaction()")
 	}
 
+	// create a single use gas pool (as we don't want the gas consumed by the inner tx to blow the block gasLimit on a participant node)
+	singleUseGasPool := new(GasPool).AddGas(innerTx.Gas())
+	used := uint64(0)
```

Comment:
`used` doesn't need to be created earlier

Reply:
Good point. Updated.